src/app/api/telegram/route.ts

```diff
@@ -0,0 +1,88 @@
+import { NextRequest,NextResponse } from 'next/server';
+import * as dotenv from "dotenv"
+import { getTelegramIntegrationFromUserId, saveTelegramIntegration, sendTelegramMessage } from '@/lib/utils';
+import { createClient } from '@supabase/supabase-js';
+import { error } from 'console';
+import { createJob, deleteJob } from '@/lib/cronjobs';
+dotenv.config();
+
+type UserState = {
+  state: string;
+  timestamp: number;
+};
+var url = process.env.NEXT_PUBLIC_SUPABASE_URL??"";
+var anon = process.env.NEXT_PUBLIC_SUPABASE_SERVICE_KEY??"";
+const supabase =  createClient(url,anon);
+const userdata = supabase.auth.admin.listUsers().then(x=>x.data);
+
+const users = userdata.then((x)=>x.users);
+
+let userStates: Record<string, UserState> = {}; 
+const STATE_EXPIRY_TIME = 60 * 60 * 24 *1000 ; 
+
+function isStateExpired(timestamp: number) {
+  return Date.now() - timestamp > STATE_EXPIRY_TIME;
+}
+
+export async function POST(request: NextRequest) {
+  try {
+    const body = await request.json();
+    const { message } = body;
+
+    if (message) {
+      const chatId = message.chat.id;
+      const text:string = message.text;
```

Comment:
Maybe this variable could be renamed to `userInput`. It could improve the readability of the code. Let me know what you think

Reply:
Yes,It is intuitive name and improves readability.